spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/registry/RdbmsUriRegistry.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.server.registry;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Types;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.sql.DataSource;
+
+import org.springframework.cloud.dataflow.server.repository.AbstractRdbmsKeyValueRepository;
+import org.springframework.cloud.dataflow.server.repository.DeploymentIdRepository;
+import org.springframework.cloud.deployer.resource.registry.UriRegistry;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.util.Assert;
+
+/**
+ * RDBMS implementation of {@link UriRegistry}.
+ *
+ * @author Ilayaperumal Gopinathan
+ */
+public class RdbmsUriRegistry extends AbstractRdbmsKeyValueRepository<String> implements UriRegistry {
+
+	public RdbmsUriRegistry(DataSource dataSource) {
+		super(dataSource, "URI_APP_", "REGISTRY", new RowMapper<String>() {
+			@Override
+			public String mapRow(ResultSet resultSet, int i) throws SQLException {
+				return resultSet.getString("NAME") + "=" + resultSet.getString("URI");
+			}
+		}, "NAME", "URI");
+	}
+
+	@Override
+	public URI find(String name) {
+		String entry = findOne(name);
+		if (entry == null) {
+			return null;
+		}
+		String[] splitEntries = entry.split("=");
+		return toUri(splitEntries[1]);
+	}
+
+	@Override
+	public Map<String, URI> findAll() {
+		Iterable<String> entries = findAllEntries();
+		Map<String, URI> map = new HashMap<>();
+		for (String entry : entries) {
+			String[] splitEntries = entry.split("=");
```

Comment:
this seems dangerous since a `URI` value *could* include an equals sign, so this should minimally provide a max value of 2